### ISSUE TITLE
remove include/rpc_generated.h 

### ISSUE THREAD

**Issue by senior7515** [smfrpc/smf#317]
currently I'm reverting a bug of 
 on #314  - once this fix is merged which is a breaking change that we need to fix. we'll need to fix the installation so it always includes the generated file in the install folder 
```
cd /home/agallego/workspace/v/build/src/v/chain_replication && ccache /bin/c++  -DSEASTAR_HAVE_DPDK -DSEASTAR_HAVE_GCC6_CONCEPTS -DXXH_PRIVATE_API -I/home/agallego/workspace/v/build/src/v/chain_replication -I/home/agallego/workspace/v/build/src/v/chain_replication/.. -I/home/agallego/workspace/v/build/src/v/chain_replication/../filesystem -I/home/agallego/workspace/v/src/v -I/home/agallego/workspace/v/build/src/v/filesystem -isystem /home/agallego/workspace/v/build/v_deps_install/include -isystem /home/agallego/workspace/v/build/v_deps_install/include/dpdk -isystem /home/agallego/workspace/v/build/v_deps_install/include/hdr -isystem /home/agallego/workspace/v/build/v_deps_install/include/re2  -std=gnu++17 -fuse-ld=gold -O2 -g -DNDEBUG   -Wall -Werror -fconcepts -Wno-unused-variable -Wno-error=maybe-uninitialized -std=gnu++17 -U_FORTIFY_SOURCE -march=native -o CMakeFiles/rpchains.dir/chain_replication_service.cc.o -c /home/agallego/workspace/v/src/v/chain_replication/chain_replication_service.cc
In file included from /home/agallego/workspace/v/build/v_deps_install/include/smf/rpc_envelope.h:9,
                 from /home/agallego/workspace/v/build/v_deps_install/include/smf/rpc_service.h:4,
                 from /home/agallego/workspace/v/build/src/v/chain_replication/chain_replication.smf.fb.h:11,
                 from /home/agallego/workspace/v/src/v/chain_replication/chain_replication_service.h:4,
                 from /home/agallego/workspace/v/src/v/chain_replication/chain_replication_service.cc:1:
/home/agallego/workspace/v/build/v_deps_install/include/smf/rpc_letter.h:9:10: fatal error: smf/rpc_generated.h: No such file or directory
 #include "smf/rpc_generated.h"
          ^~~~~~~~~~~~~~~~~~~~~

```

**Comment by noahdesu** [comment]
what's the issue here? i believe i removed this in the cmake-ng pr. it does need to be installed from the binary directory.

**Comment by senior7515** [comment]
fixed in #314  

The issue is that we need to have an `install()` for the generated file in a transitive cmake project.
